### ISSUE TITLE
Fix `declaration-property-value-keyword-no-deprecated` false positives

### ISSUE THREAD

**Issue by Stanzilla** [stylelint/stylelint#8327]
### What minimal example or steps are needed to reproduce the bug?

```typescript
const SensitiveContentInner = styled.div`
  background-color: ${palette(({ background }) => background.primary)};
  border: 1px solid ${palette(({ border }) => border.primary)};
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100%;
  width: 100%;
`;

const RewardNoticeHolder = styled.div`
  position: absolute;
  background: ${palette(({ background }) => background.tertiary)};
  border-radius: ${shape(({ wrapperBorderRadius }) => wrapperBorderRadius)};
  border-top-right-radius: 0;
  padding: ${spacing(0.5)} ${spacing(1)};
  text-align: right;
  &::before {
    content: '';
    position: absolute;
    right: 0;
    bottom: 100%;
    width: 0;
    height: 0;
    border-style: solid;
    border-width: 0 0 12px 16px;
    border-color: transparent transparent
      ${palette(({ background }) => background.tertiary)} transparent;
  }
`;
```

### What minimal configuration is needed to reproduce the bug?

```js
/** @type {import('stylelint').Config} */
module.exports = {
  extends: ['stylelint-config-recommended'],
  customSyntax: 'postcss-styled-syntax',
  plugins: ['stylelint-plugin-logical-css'],
  rules: {
    'plugin/use-logical-properties-and-values': [
      true,
      {
        severity: 'warning',
        ignore: [
          'height',
          'max-height',
          'min-height',
          'width',
          'max-width',
          'min-width',
          'contain-intrinsic-height',
          'contain-intrinsic-width',
          'overflow-x',
          'overflow-y',
          'overscroll-behavior-x',
          'overscroll-behavior-y',
          'box-orient: vertical',
          'box-orient: horizontal',
          '-webkit-box-orient: vertical',
          '-webkit-box-orient: horizontal',
          'resize: horizontal',
          'resize: vertical'
        ]
      }
    ],
    'plugin/use-logical-units': [true, { severity: 'warning' }],
    'value-no-vendor-prefix': true,
    'property-no-vendor-prefix': true,
    'no-empty-source': null,
    'unit-no-unknown': [
      true,
      {
        ignoreUnits: ['svh']
      }
    ],
    'selector-type-no-unknown': [
      true,
      {
        ignoreTypes: [
          'Texteffecttext-shadow',
          'znipe-multi-view-player',
          'znipe-single-view-player',
          'dropFile'
        ]
      }
    ],
    'property-no-unknown': [
      true,
      {
        ignoreProperties: [
          'margin-vertical',
          'margin-horizontal',
          'padding-horizontal',
          'padding-vertical',
          'tint-color',
          'resize-mode',
          'aspect-ratio',
          'shadow-radius',
          'shadow-color',
          'shadow-offset',
          'shadow-opacity',
          'theme'
        ]
      }
    ]
  }
};
```

### How did you run Stylelint?

VSCode Extension

### Which Stylelint-related dependencies are you using?

```json
  "stylelint": "^16.12.0",
    "stylelint-config-recommended": "^14.0.1",
    "stylelint-formatter-gitlab": "^1.0.2",
    "stylelint-plugin-logical-css": "^1.2.1",
```

### What did you expect to happen?

No errors

### What actually happened?

The following problems were reported:

```
Expected "${palette(({ background }) => background.primary)}" to be "canvas" (declaration-property-value-keyword-no-deprecated)Stylelint[declaration-property-value-keyword-no-deprecated](https://stylelint.io/user-guide/rules/declaration-property-value-keyword-no-deprecated)
```

```
Expected "transparent transparent
      ${palette(({ background }) => background.tertiary)} transparent" to be "canvas" (declaration-property-value-keyword-no-deprecated)
```

**Comment by Mouvedia** [comment]
Hello @Stanzilla,

There's 2 different problems: 

1. `background-color: ${palette(({ background }) => background.primary)};`
2. `border-color: transparent transparent ${palette(({ background }) => background.tertiary)} transparent;`

I am not sure how to fix the first one just yet but I see what's wrong in the second one.

**Comment by Stanzilla** [comment]
thank you!

**Comment by Mouvedia** [comment]
> `ignoreUnits: ['svh']`

BTW [`svh` was added](https://github.com/stylelint/stylelint/pull/5970) in 2022.

**Comment by Stanzilla** [comment]
@Mouvedia hey there, I just tested 16.14.1 and it looks like the `background` case is not quite fixed.

Code
```css
  background-color: ${palette(({ background }) => background.primary)};
```

Expected "background" to be "canvas" (declaration-property-value-keyword-no-deprecated)

**Comment by Mouvedia** [comment]
I see what's going on. It was fixed for certain cases but not all.
I should have a fix ready soon.

**Comment by Stanzilla** [comment]
> I see what's going on. It was fixed for certain cases—e.g. Sass—but not all cases. I should have a fix ready soon.

Awesome, thank you!

**Comment by Mouvedia** [comment]
@Stanzilla with #8352 closed you might be able to test the fix before the release.
